File: Java/InsertionSort.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.InputMismatchException;
import java.util.Scanner;

public class InsertionSort {
    public static void main(String[] args) {
        ArrayList<Integer> arr = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter size of the list: ");
        try {
            int size = scanner.nextInt();

            if (size <= 0) {
                System.out.println("Size of list can not be negative or zero");
                scanner.close();
                return;
            }
            System.out.println("Enter the elements: ");
            for (int i = 0; i < size; i++) {
                int el = scanner.nextInt();
                arr.add(el);
            }
            System.out.println("INPUT LIST: ");
            for (int num : arr)
                System.out.print(num + "\t");
            insertionSort(arr);
            System.out.println("\nSORTED LIST: ");
            for (int num : arr)
                System.out.print(num + "\t");
        } catch (InputMismatchException e) {
            System.out.println("INPUT MUST BE IN NUMBERS");
        }
        scanner.close();
    }

    public static void insertionSort(ArrayList<Integer> arr) {
        int size = arr.size();
        for (int i = 0; i < size; i++) {
            int j = i;
            while (j > 0 && arr.get(j - 1) > arr.get(j)) {
                Collections.swap(arr, j, j - 1);
                j--;
            }
        }
    }

}

File: CPP/InsertionSort.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>
#define vec vector<ll>
#define ll long long int
using namespace std;

int insertionSort(vec &arr)
{
    int size = arr.size();
    for (int i = 0; i < size; i++)
    {
        int j = i;
        while (j > 0 && arr[j - 1] > arr[j])
        {

            swap(arr[j - 1], arr[j]);
            j--;
        }
    }
}

int inputNumber()
{
    int num;
    cin >> num;
    if (cin.fail())
    {
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
        throw runtime_error("INPUT MUST BE A NUMBER");
    }
    return num;
}

void printArray(vec arr)
{
    cout << '\t';
    for (auto &i : arr)
        cout << i << '\t';
    cout << endl;
}

int main()
{
    system("cls");
    //     ios_base::sync_with_stdio(false);
    //     cin.tie(NULL);
    // #ifndef ONLINE_JUDGE
    //     freopen("../input.txt", "r", stdin);
    //     freopen("../output.txt", "w", stdout);
    // #endif

    vec arr;
    int size, testCases;
    testCases = inputNumber();

    while (testCases--)
    {
        try
        {
            ll element;
            size = inputNumber();
            arr.clear(); // Clearing array so that every iteration have a new one
            if (size <= 0)
                throw runtime_error("SIZE SHOULD BE GREATER THAN ZERO");
            for (int i = 0; i < size; i++)
                element = inputNumber(), arr.push_back(element);
            cout << "INPUT ARRAY: ";
            printArray(arr);
            insertionSort(arr);
            cout << "SORTED ARRAY: ";
            printArray(arr);
        }
        catch (runtime_error &e)
        {
            cout << "ERROR: " << e.what() << endl;
        }
        cout << endl;
    }

    return 0;
}